lib/Mojolicious/Controller.pm

```diff
@@ -262,8 +278,15 @@ sub signed_cookie {
   # Request cookie
   return $self->every_signed_cookie($name)->[-1] unless defined $value;
 
+  my $app     = $self->app;
+  my $secret  = $app->secrets->[0];
+  my $moniker = $app->moniker;
+
+  Carp::croak 'Your secret passphrase must be changed to set signed cookies (see FAQ for more)'
+    if !$ENV{MOJO_ALLOW_INSECURE_SECRET} and (!length $secret or $secret eq $moniker);
```

Comment:
Would it make sense to move this check into `$app->session`?

Reply:
yes, sorry, github threw away my comment, and i crankly re-typed it wrong.

I mean should it be in `$app->secrets`

Reply:
That attribute may be accessed by applications that don't actually do anything with it, such as Mojolicious::Plugin::Mount. But perhaps we can work around that more easily than the problems with this approach.